use_cases/eluc/tests/test_nsga2_utils.py

```diff
@@ -41,3 +42,29 @@ def test_distance_calculation(self):
         nsga2_utils.calculate_crowding_distance(shuffled_front)
         for candidate, tgt in zip(shuffled_front, shuffled_tgts):
             self.assertAlmostEqual(candidate.distance, tgt)
+
+    def manual_two_obj_dominate(self, candidate1: Candidate, candidate2: Candidate) -> bool:
+        """
+        Manually test all cases of domination for 2 objectives.
+        For candidate 1 to dominate candidate 2 it must:
+            - Have a lower value in at least one objective
+            - Have lower or equal values in all the rest
+        """
+        if (candidate1.metrics[0] < candidate2.metrics[0]) and (candidate1.metrics[1] <= candidate2.metrics[1]):
+            return True
+        if (candidate1.metrics[0] <= candidate2.metrics[0]) and (candidate1.metrics[1] < candidate2.metrics[1]):
+            return True
+        return False
+
+    def test_domination_two_obj(self):
+        """
+        Tests domination works in all possible cases.
+        Get all combinations of pairs of values [0, 1, 2] for each objective and tests against the manual checker.
+        """
+        for comb in itertools.combinations([0, 1, 2], 4):
+            candidate1 = Candidate(16, 16, 16)
+            candidate2 = Candidate(16, 16, 16)
+            candidate1.metrics = [comb[0], comb[1]]
+            candidate2.metrics = [comb[2], comb[3]]
+            self.assertEqual(nsga2_utils.dominates(candidate1, candidate2), self.manual_two_obj_dominate(candidate1, candidate2))
```

Comment:
Added a test that manually checks all combinations of 2 candidates with 2 objectives against a hard-coded domination function.